pkg/cloudprovider/cloud.go

```diff
@@ -185,4 +185,14 @@ type Zone struct {
 type Zones interface {
 	// GetZone returns the Zone containing the current failure zone and locality region that the program is running in
 	GetZone() (Zone, error)
+
```

Comment:
I would add a comment here saying that GetZone() originally was always run from the kubelet, and this method used the local metadata service on the node running in the cloudprovider to obtain this information. Therefore, there was no arguments needed to be passed in. 

Now that we are using an external cloud controller manager, that does not run on every node, we need to identify the node whose zone information we are obtaining. Hence there is a need for identifying the nodes in these specific ways.